rust/cx-api/build.rs

```diff
@@ -12,26 +12,76 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::path::PathBuf;
+use git2::Repository;
+use std::{
+    fs,
+    io,
+    path::{
+        Path,
+        PathBuf,
+    },
+};
+const PROTOS_DIR: &str = "proto";
+
+/// Copy files from source to destination recursively.
+pub fn copy_recursively(source: impl AsRef<Path>, destination: impl AsRef<Path>) -> io::Result<()> {
+    if destination.as_ref().exists() {
+        fs::remove_dir_all(&destination)?;
+    }
+    fs::create_dir_all(&destination)?;
+    for entry in fs::read_dir(source)? {
+        let entry = entry?;
+        let filetype = entry.file_type()?;
+        if filetype.is_dir() {
+            copy_recursively(entry.path(), destination.as_ref().join(entry.file_name()))?;
+        } else {
+            fs::copy(entry.path(), destination.as_ref().join(entry.file_name()))?;
+        }
+    }
+    Ok(())
+}
+
+fn is_dev_repository() -> bool {
+    if let Ok(repo) = Repository::discover(std::env::var("CARGO_MANIFEST_DIR").unwrap()) {
```

Comment:
use chaining so it doesn't crash and burn if CARGO_MANIFEST_DIR isn't set

Reply:
Done :-)